.github/workflows/appimage.yml

```diff
@@ -26,12 +26,25 @@ jobs:
     timeout-minutes: 30
 
     steps:
-    - name: Check out code in viam-cartographer directory
+    - name: Check out main code
+      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
       uses: actions/checkout@v3
       with:
         submodules: recursive
         path: viam-cartographer
+      
+    - name: Check out PR branch code
+      if: github.event_name == 'pull_request_target'
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
1. When will `github.event_name == 'pull_request_target'` evaluate to true?

2. What is this ref doing?

Reply:
pull_request_target is when this branch (normally "main") is the target of a PR. E.g. when someone wants to merge to main.

The ref is saying "check out the PR's code" (which may be untrusted) instead of the default of checking out the code in the context of the workflow (e.g. "main")